javalib/src/main/scala/java/util/Arrays.scala

```diff
@@ -82,61 +203,48 @@ object Arrays {
     sortRangeImpl[Double](a, fromIndex, toIndex)
 
   @noinline def sort(a: Array[AnyRef]): Unit =
-    sortAnyRefImpl(a)
+    sortImpl(a)
 
   @noinline def sort(a: Array[AnyRef], fromIndex: Int, toIndex: Int): Unit =
-    sortRangeAnyRefImpl(a, fromIndex, toIndex)
+    sortRangeImpl(a, fromIndex, toIndex)
 
-  @noinline def sort[T <: AnyRef](array: Array[T], comparator: Comparator[_ >: T]): Unit = {
-    implicit val ord = toOrdering(comparator).asInstanceOf[Ordering[AnyRef]]
-    sortAnyRefImpl(array.asInstanceOf[Array[AnyRef]])
-  }
+  @noinline def sort[T <: AnyRef](array: Array[T], comparator: Comparator[_ >: T]): Unit =
+    sortImpl(array)(new ComparatorArrayOps(comparator), new TemplateArrayOps(array))
```

Comment:
It must have crossed my mind at some point. But there is one problem with that: the dispatch for `Comparable.compareTo(Comparable)`, used for primitives, is more expensive than it should be when sorting an array of primitives. At least with `ArrayCompareOps`, the cost is always one polymorphic method call away. With the dispatch, it's several type tests in a row.

Whether that's a strong enough argument is unclear. It would be an additional performance hit for `sort` on arrays of primitives. This wouldn't be an issue if we could inline/code-generate `sort` for primitives.